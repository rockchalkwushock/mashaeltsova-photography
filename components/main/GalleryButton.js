import Link from 'next/link'
import PropTypes from 'prop-types'

import { styles } from '../../lib'

const GalleryButton = ({ text, url }) => (
  <Link prefetch href={url}>
    <button className="gallery-btn">
      {text}
      <style jsx>{`
        .gallery-btn {
          align-self: center;
          background-color: ${styles.colors.peach};
          border: 0;
          box-shadow: 0 0 20px rgba(54, 69, 79, 0.7);
          color: rgba(223, 116, 74, 0.9);
          cursor: pointer;
          font-family: ${styles.fonts.courgette};
          height: 2em;
          letter-spacing: 0.25em;
          outline: 0;
          padding: 0.5em;
          position: absolute;
        }
        .gallery-btn:focus,
        .gallery-btn:hover {
          align-self: center;
          background-color: ${styles.colors.peach};
          border: 0;
          color: ${styles.colors.warm};
          font-size: 1.4em;
          height: 2em;
          outline: 0;
          -webkit-transition: 0.2s ease;
          -moz-transition: 0.2s ease;
          transition: 0.2s ease;
        }
      `}</style>
    </button>
  </Link>
)

GalleryButton.propTypes = {
  text: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

export default GalleryButton
