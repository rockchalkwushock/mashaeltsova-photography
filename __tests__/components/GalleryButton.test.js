import React from 'react'
import { render } from 'enzyme'

import GalleryButton from '../../components/main/GalleryButton'

describe('Component: GalleryButton', () => {
  test('Renders without exploding', () => {
    const tree = render(<GalleryButton text="go here" url="/" />)
    expect(tree).toMatchSnapshot()
  })
})
