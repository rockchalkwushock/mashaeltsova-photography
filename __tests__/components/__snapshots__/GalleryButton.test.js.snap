// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: GalleryButton Renders without exploding 1`] = `
<button
  class="jsx-4102330989 gallery-btn"
>
  go here
</button>
`;
